pkg/scale/decode.go

```diff
@@ -121,46 +119,94 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding big integer: %w", err)
```

Comment:
nit maybe we could return early directly for every error cases? Same for below

```suggestion
			return fmt.Errorf("decoding big integer: %w", err)
```